.github/workflows/sycl-nightly.yml

```diff
@@ -15,7 +15,14 @@ jobs:
     with:
       build_cache_root: "/__w/"
       build_artifact_suffix: default
-      build_configure_extra_args: '--hip --cuda'
+      build_configure_extra_args:
```

Comment:
Should we build precommit linux with the same flags (It looks like we don't but maybe I can't read)

Reply:
I added this for more diversity i.e. pre-commit is static build (llvm config default), and then nightly testing gets exercised with a shared config. I'm not married to it, but I figured it was any easy way to increase coverage

Reply:
One issue we actually do see relatively frequently is someone breaking the shared library build (usually missing a library dep in the cmake file) and today that is usually caught in post commit because that does shared lib build. With this, we would only catch it in the nightly. 

However I think it's reasonable to assume that most developers are using the default config (configure.py) and assuming we have someone guarneteed to be checking the nightly and quickly report and/or fix issues, I don't have a huge problem with this

Reply:
> We could swap it round, so pre-commit testing catches the sharelib issues?

I wonder if it's possible to break static library build and not shared library build. If that's extremely unlikely such that either both are broken or shlib is broken, doing shlib in precommit would be fine with me.

Reply:
We probably should just drop "sharedlib" build configuration. The primary use case for this configuration is to speed-up incremental build of LLVM libraries. This build configuration is not use in the production environment - it's used only by LLVM developers. If everyone uses standard build, there is no point to build "sharedlib" configuration in CI.

Reply:
Personally I use it rather frequently and I would be annoyed if it was failing and not caught in CI.

Reply:
Our downstream uses it exclusively. It would be very bad if the issues will only be discovered when syncing downstream with upstream and not in upstream CI.